src/main/java/com/massivecraft/factions/zcore/persist/MemoryBoard.java

```diff
@@ -158,7 +159,18 @@ public void unclaimAllInWorld(String factionId, World world) {
     }
 
     public void clean(String factionId) {
-        flocationIds.removeFaction(factionId);
+        if (LWCFeatures.getEnabled() && P.p.getConfig().getBoolean("lwc.reset-locks-unclaim", false)) {
+            Iterator<Entry<FLocation, String>> iter = flocationIds.entrySet().iterator();
+            while (iter.hasNext()) {
+                Entry<FLocation, String> entry = iter.next();
+                if (entry.getValue().equals(factionId)) {
+                    LWCFeatures.clearAllChests(entry.getKey());
+                    iter.remove();
+                }
+            }
+        } else {
+            flocationIds.removeFaction(factionId);
```

Comment:
Unless I missed something, I believe running this only in the `else` means that the factionToLandMap will still contain that faction's old places.

Reply:
To be honest I don't know the answer to this. I mostly added the old code back without much regard.
If you could point me in the right direction I can try to fix this.